com.gitlab.sat_metalab.Splash.json

```diff
@@ -1,7 +1,7 @@
 {
     "app-id": "com.gitlab.sat_metalab.Splash",
     "runtime": "org.freedesktop.Platform",
-    "runtime-version": "22.08",
+    "runtime-version": "21.08",
```

Comment:
what are the problems you are encountering?

Reply:
I just get a compilation error:
```
/run/build/splash/src/utils/jsonutils.cpp: In function ‘bool Splash::Utils::checkAndUpgradeConfiguration(Json::Value&)’:
/run/build/splash/src/utils/jsonutils.cpp:129:40: error: class template argument deduction failed:
  129 |             "weightedCalibrationPoints"};
      |                                        ^
/run/build/splash/src/utils/jsonutils.cpp:129:40: error: no matching function for call to ‘array(const char [7], const char [10], const char [5], const char [5], const char [14], const char [11], const char [11], const char [8], const char [5], const char [15], const char [10], const char [8], const char [8], const char [5], const char [26])’
In file included from /usr/include/c++/12.1.0/bits/stl_map.h:63,
                 from /usr/include/c++/12.1.0/map:61,
                 from /app/include/json/value.h:17,
                 from /app/include/json/json.h:10,
                 from /run/build/splash/src/././utils/jsonutils.h:32,
                 from /run/build/splash/src/utils/jsonutils.cpp:1:
/usr/include/c++/12.1.0/tuple:1595:45: note: candidate: ‘template<class _Tp, long unsigned int _Nm> array()-> std::array<_Tp, _Nm>’
 1595 |   template<typename _Tp, size_t _Nm> struct array;
      |                                             ^~~~~
/usr/include/c++/12.1.0/tuple:1595:45: note:   template argument deduction/substitution failed:
/run/build/splash/src/utils/jsonutils.cpp:129:40: note:   candidate expects 0 arguments, 15 provided
  129 |             "weightedCalibrationPoints"};
      |                                        ^
/usr/include/c++/12.1.0/tuple:1595:45: note: candidate: ‘template<class _Tp, long unsigned int _Nm> array(std::array<_Tp, _Nm>)-> std::array<_Tp, _Nm>’
 1595 |   template<typename _Tp, size_t _Nm> struct array;
      |                                             ^~~~~
/usr/include/c++/12.1.0/tuple:1595:45: note:   template argument deduction/substitution failed:
/run/build/splash/src/utils/jsonutils.cpp:129:40: note:   mismatched types ‘std::array<_Tp, _Nm>’ and ‘const char*’
  129 |             "weightedCalibrationPoints"};
      |                                        ^
make[2]: *** [src/CMakeFiles/splash-0.9.dir/build.make:916: src/CMakeFiles/splash-0.9.dir/utils/jsonutils.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:447: src/CMakeFiles/splash-0.9.dir/all] Error 2
make: *** [Makefile:166: all] Error 2
Error: module splash: Child process exited with code 2
```
It is related to the structure located here: https://gitlab.com/sat-mtl/tools/splash/splash/-/blob/master/src/utils/jsonutils.cpp#L115

Strangely I've not be able to reproduce this anywhere else, even using GCC 12.2 on Archlinux.

Reply:
Would it be possible to merge this MR without fixing this right now?

Reply:
For your information, I tried building with runtime 22.08 on the CI and it failed too. I'll have to find a way to reproduce this in a setup which allows for testing fixes, as it does not happen anywhere else (be it Ubuntu or Archlinux).